src/io/hdf5_writer.cpp

```diff
@@ -180,10 +176,10 @@ void HDF5Writer::close() {
     if (spikes_attr_type_) {
         H5Tclose(spikes_attr_type_);
         spikes_attr_type_ = 0;
-    }
-    if (file_) {
-        H5Fclose(file_);
-        file_ = 0;
+        if (file_) {
```

Comment:
At a glance, shouldn't all ranks close the file, independently of this `spikes_attr_type_` ?